sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubJsonClient.java

```diff
@@ -135,7 +135,7 @@ public int publish(TopicPath topic, List<OutgoingMessage> outgoingMessages)
     for (OutgoingMessage outgoingMessage : outgoingMessages) {
       PubsubMessage pubsubMessage = new PubsubMessage().encodeData(outgoingMessage.elementBytes);
 
-      Map<String, String> attributes = pubsubMessage.getAttributes();
+      Map<String, String> attributes = outgoingMessage.attributes;
```

Comment:
yes,  message attributes always be null, except when using `timestampLabel` and `idLabel`.